### ISSUE TITLE
init: symbols for functions with pre-code pools have their address set to the pool address, resulting in delink InvalidStart error

### ISSUE THREAD

**Issue by LagoLunatic** [AetiasHax/ds-decomp#7]
Affects: Castlevania: Order of Ecclesia (YR9E00) and Castlevania: Portrait of Ruin (ACBE00)

Running init for OoE appears to work:
<details open>
<summary><h3>OoE init</h3></summary>
<pre>
>dsd init --rom-config ./orig/YR9E00/config.yaml --output-path ./config/YR9E00 --build-path ./build/YR9E00 
[2024-11-12T21:42:01Z INFO  ds_rom::rom::rom] Loading ROM from ./orig/YR9E00/config.yaml
[2024-11-12T21:42:01Z INFO  ds_decomp::analysis::functions] Function at 020163f4 was adjusted to include pre-code constant pool at 020163e0
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x02023e30 in ARM9 main to 0x0202407c goes to middle of function 'func_02024074' at 0x02024074, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ff8cac in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ff8cd8 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffdd38 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffde10 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffde60 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffdea0 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffdee0 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffdf28 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffdf60 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01ffdf98 in ITCM to 0x01ff8b6c goes to middle of function 'func_01ff8b3c' at 0x01ff8b3c, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01fff0f0 in ITCM to 0x01fff1cc goes to middle of function 'func_01fff0d8' at 0x01fff0d8, adding an external label symbol
[2024-11-12T21:42:01Z WARN  ds_decomp::analysis::data] Function call from 0x01fff0f4 in ITCM to 0x01fff20c goes to middle of function 'func_01fff0d8' at 0x01fff0d8, adding an external label symbol
</pre>
</details>

However, if you then run `delink` you get the following error:
```
>dsd delink --config-path ./config/YR9E00/arm9/config.yaml 
[2024-11-12T22:11:53Z INFO  ds_rom::rom::rom] Loading ROM from ./config/YR9E00/arm9\../../../orig/YR9E00/config.yaml
Error: function func_020163f4 could not be analyzed: InvalidStart
```
Looking at the symbols.txt generated by `init` reveals the following line:
```
func_020163f4 kind:function(arm,size=0x294) addr:0x20163e0
```
So it seems that `init` now understands that the function starts at 020163F4 and uses a pool at 020163E0. But it incorrectly sets the address of the symbol to the pool's address instead of the function's address. If I manually edit symbols.txt to fix the address, then delink runs without errors for OoE.

Related: #1

**Comment by AetiasHax** [comment]
Fixed, see v0.2.2! I also worked around some odd behaviors of mwldarm, so OoE will now match after linking.

**Comment by LagoLunatic** [comment]
Works perfectly now, thanks!
